test/Faker/Provider/InternetTest.php

```diff
@@ -108,6 +108,16 @@ public function testIpv4()
         $this->assertNotFalse(filter_var($this->faker->ipv4(), FILTER_VALIDATE_IP, FILTER_FLAG_IPV4));
     }
 
+    public function testIpv4NotLocalNetwork()
+    {
+        $this->assertNotEquals('1.0.0.0', long2ip(ip2long($this->faker->ipv4()) & (-1 << (32 - 8))));
```

Comment:
It shifts away the last three bits (B, C,& D from A.B.C.D), so only the first bit (A) remains.
This way we can compare it.
Probably there is a more efficient way, but I'm no binary calculation hero :smirk_cat: